firmware/TempoHandler.h

```diff
@@ -49,38 +50,33 @@ class TempoHandler
     inline void setHandleAlignEvent(void (*fptr)()) {
       tAlignCallback = fptr;
     }
-    void setPPQN(int ppqn) {
-      _ppqn = ppqn;
-    }
     void update() {
-      // Determine which source is selected for tempo
-      uint8_t new_source = _source;
-
-      if(Sync::detect()) {
-        new_source = TEMPO_SOURCE_SYNC;
-      } else if (_midi_clock_received_flag) { 
-        new_source = TEMPO_SOURCE_MIDI;
-      } else {
-        new_source = TEMPO_SOURCE_INTERNAL;
-      }
-
-      if (new_source != _source) {
-        _source = new_source;
-        _midi_clock_received_flag = 0;
-        tempo.reset();
+      static const unsigned second = 1000 * 1000;
+      static const unsigned timeout = 2 * second;
```

Comment:
Arbitrarily chose 2 seconds as the timeout before we reset to internal clock.

Reply:
we shouldn't fall back to the internal clock after a timeout. The only way to fall back should be to unplug and replug the sync in, or to power cycle the unit

Reply:
removed